File: app/src/main/java/tech/destinum/machines/data/local/ViewModel/IncomeViewModel.java
package tech.destinum.machines.data.local.ViewModel;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;
import android.database.Cursor;


import java.util.List;

import javax.inject.Inject;

import io.reactivex.Completable;
import io.reactivex.Flowable;
import io.reactivex.Maybe;
import io.reactivex.Observable;
import io.reactivex.Single;
import tech.destinum.machines.data.MachinesDB;
import tech.destinum.machines.data.local.POJO.Income;

public class IncomeViewModel extends ViewModel{

    private MachinesDB machinesDB;

    @Inject
    public IncomeViewModel(MachinesDB machinesDB) {
        this.machinesDB = machinesDB;
    }

    public Flowable<Double> getIncomeOfMachine(long id){
        return machinesDB.getIncomeDAO().getIncomeOfMachine(id);
    }

    public Completable addIncome(long date, String note, Double money, Long machines_id, int month){
        return Completable.fromAction(() -> machinesDB.getIncomeDAO().addIncome(new Income(date, note, money, machines_id, month)));
    }

    public Flowable<Double> getTotalMonth(long machines_id, int month){
        return machinesDB.getIncomeDAO().getTotalMonth(machines_id, month);
    }

    public Flowable<List<Income>> getAllIncomesOfMachine(long machines_id){
        return machinesDB.getIncomeDAO().getInfoOfMachine(machines_id);
    }

    public LiveData<List<Income>> getLiveDataList(long machines_id){
        return machinesDB.getIncomeDAO().getLiveDataList(machines_id);
    }

    public Single<Cursor> getCursor(){
        return Single.fromCallable(() -> machinesDB.getIncomeDAO().getCursor());
    }

    public Single<Cursor> getCursorByID(long machines_id){
        return Single.fromCallable(() -> machinesDB.getIncomeDAO().getCursorByID(machines_id));
    }

    public long deleteIncomeByID(long id){
        return machinesDB.getIncomeDAO().deleteIncomeByID(id);
    }

    public Completable updateIncomeByID(long id, String note, double money){
        return Completable.fromCallable(() -> machinesDB.getIncomeDAO().updateIncomeByID(id, note, money));
    }

    public Flowable<Double> totalObtained(){
        return machinesDB.getIncomeDAO().totalObtained();
    }
}

File: app/src/main/java/tech/destinum/machines/Data/Local/ViewModel/IncomeViewModel.kt
package tech.destinum.machines.Data.Local.ViewModel

import android.database.Cursor
import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel

import javax.inject.Inject

import io.reactivex.Completable
import io.reactivex.Flowable
import io.reactivex.Single
import tech.destinum.machines.Data.MachinesDB
import tech.destinum.machines.Data.Local.Entities.Income

class IncomeViewModel @Inject
constructor(private val machinesDB: MachinesDB) : ViewModel() {

    val cursor: Single<Cursor>
        get() = Single.fromCallable { machinesDB.incomeDAO.cursor }

    fun getIncomeOfMachine(id: Long): Flowable<Double> {
        return machinesDB.incomeDAO.getIncomeOfMachine(id)
    }

    fun addIncome(date: Long, note: String, money: Double, machines_id: Long, month: Int): Completable {
        return Completable.fromAction { machinesDB.incomeDAO.addIncome(Income(date, note, money, machines_id, month)) }
    }

    fun getTotalMonth(machines_id: Long, month: Int): Flowable<Double> {
        return machinesDB.incomeDAO.getTotalMonth(machines_id, month)
    }

    fun getAllIncomesOfMachine(machines_id: Long): Flowable<List<Income>> {
        return machinesDB.incomeDAO.getInfoOfMachine(machines_id)
    }

    fun getLiveDataList(machines_id: Long): LiveData<List<Income>> {
        return machinesDB.incomeDAO.getLiveDataList(machines_id)
    }

    fun getCursorByID(machines_id: Long): Single<Cursor> {
        return Single.fromCallable { machinesDB.incomeDAO.getCursorByID(machines_id) }
    }

    fun deleteIncomeByID(id: Long): Long {
        return machinesDB.incomeDAO.deleteIncomeByID(id).toLong()
    }

    fun updateIncomeByID(id: Long, note: String, money: Double): Completable {
        return Completable.fromCallable { machinesDB.incomeDAO.updateIncomeByID(id, note, money) }
    }

    fun totalObtained(): Flowable<Double> =  machinesDB.incomeDAO.totalObtained()

    fun getTotalMonth(month: Int): Single<Double> = machinesDB.incomeDAO.getTotalMonth(month)
}
